homeassistant/components/history/__init__.py

```diff
@@ -335,35 +335,41 @@ async def get(self, request, datetime=None):
 
         hass = request.app["hass"]
 
-        result = await hass.async_add_job(
-            get_significant_states,
-            hass,
-            start_time,
-            end_time,
-            entity_ids,
-            self.filters,
-            include_start_time_state,
-            significant_changes_only,
-        )
-        result = list(result.values())
-        if _LOGGER.isEnabledFor(logging.DEBUG):
-            elapsed = time.perf_counter() - timer_start
-            _LOGGER.debug("Extracted %d states in %fs", sum(map(len, result)), elapsed)
-
-        # Optionally reorder the result to respect the ordering given
-        # by any entities explicitly included in the configuration.
-        if self.use_include_order:
-            sorted_result = []
-            for order_entity in self.filters.included_entities:
-                for state_list in result:
-                    if state_list[0].entity_id == order_entity:
-                        sorted_result.append(state_list)
-                        result.remove(state_list)
-                        break
-            sorted_result.extend(result)
-            result = sorted_result
-
-        return await hass.async_add_job(self.json, result)
+        def _sorted_significant_states_json():
```

Comment:
Should we make this function a helper method so we don't need to define it every request?

Reply:
I don't think the overhead is anything to talk about in python. It's also makes it much larger with all parameters to pass in.

Reply:
The internal function is easier to read, but it does perform as well (its close).   The history api gets pounded with requests if you have a lot of gauges so anything we can get out of it is a nice improvement.   I went back to the original solution as the variable passing isn't in another file and is right below so it doesn't make it significantly harder to maintain.